File: backend-webflux/src/main/java/com/ups/upsglam_backend/dto/UserSearchResult.java
package com.ups.upsglam_backend.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.UUID;

@Data
@AllArgsConstructor
public class UserSearchResult {
    private UUID id;
    private String username;

    @JsonProperty("avatar_url")
    private String avatarUrl;

    @JsonProperty("is_following")
    private boolean isFollowing;
}

File: ups_glam/lib/core/models/user_search_result.dart
class UserSearchResult {
  final String id;
  final String username;
  final String? avatarUrl;
  final bool isFollowing;

  const UserSearchResult({
    required this.id,
    required this.username,
    this.avatarUrl,
    required this.isFollowing,
  });

  factory UserSearchResult.fromJson(Map<String, dynamic> j) => UserSearchResult(
        id: j['id'] as String,
        username: j['username'] as String,
        avatarUrl: j['avatar_url'] as String?,
        isFollowing: j['is_following'] as bool? ?? false,
      );

  UserSearchResult copyWith({bool? isFollowing}) => UserSearchResult(
        id: id,
        username: username,
        avatarUrl: avatarUrl,
        isFollowing: isFollowing ?? this.isFollowing,
      );
}
